designer/client/src/components/toolbars/search/AdvancedSearchFilters.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useMemo, useRef } from "react";
+import { Button, Box, Typography } from "@mui/material";
+import { useTranslation } from "react-i18next";
+import { SearchLabeledInput } from "../../sidePanels/SearchLabeledInput";
+import { SearchLabel } from "../../sidePanels/SearchLabel";
+
+const transformInput = (input: string, fieldName: string) => {
+    return input === "" ? "" : `${fieldName}:(${input})`;
+};
+
+function extractSimpleSearchQuery(text: string): string {
+    const regex = /(\w+):\(([^)]*)\)/g;
+    let match: RegExpExecArray | null;
+    let lastIndex = 0;
+
+    while ((match = regex.exec(text)) !== null) {
+        lastIndex = regex.lastIndex;
+    }
+
+    const rest = text.slice(lastIndex).trim();
+
+    return rest;
+}
+
+export function AdvancedSearchFilters({
+    filter,
+    setFilter,
+    setCollapsedHandler,
+}: {
+    filter: string;
+    setFilter: React.Dispatch<React.SetStateAction<string>>;
+    setCollapsedHandler: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
+    const { t } = useTranslation();
+    const idRef = useRef<HTMLInputElement>(null);
+    const descriptionRef = useRef<HTMLInputElement>(null);
+    const paramNameRef = useRef<HTMLInputElement>(null);
+    const paramValueRef = useRef<HTMLInputElement>(null);
+    const outputValueRef = useRef<HTMLInputElement>(null);
+    const typeRef = useRef<HTMLInputElement>(null);
+    const edgeExpressionRef = useRef<HTMLInputElement>(null);
+
+    const displayNames = useMemo(
+        () => ({
+            id: t("panels.search.field.id", "Name"),
+            description: t("panels.search.field.description", "Description"),
+            type: t("panels.search.field.type", "Type"),
+            paramName: t("panels.search.field.paramName", "Label"),
+            paramValue: t("panels.search.field.paramValue", "Value"),
+            outputValue: t("panels.search.field.outputValue", "Output"),
+            edgeExpression: t("panels.search.field.edgeExpression", "Edge"),
+        }),
+        [t],
+    );
+
+    console.log(displayNames);
+
+    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        const formData = new FormData(event.currentTarget);
+
+        console.log(formData);
+
+        const transformedInputs = Array.from(formData.entries())
+            .map(([fieldName, fieldValue]) => {
+                const input = (fieldValue as string) || "";
+                return transformInput(input, fieldName);
+            })
+            .filter((input) => input !== "");
+
+        const finalText = transformedInputs.join(" ").trim() + " " + extractSimpleSearchQuery(filter);
+
+        setFilter(finalText);
+    };
+
+    const handleCancel = () => {
```

Comment:
Is it expected behavior, that when the filter is added and we want to clear it, the form is closed?


https://github.com/user-attachments/assets/ccb5a3a1-c11d-402a-915f-818bad33124d

Reply:
Yeah, I think it in a sense reasonable to expect that if someone want to clear filers then he wants to remove them all. If user would want to edit filter then he is able to do it on his own without clearing all the filters.